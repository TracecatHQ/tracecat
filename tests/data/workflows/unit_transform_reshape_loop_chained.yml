title: Chained reshape loop
description: Test that we can chain reshaping for loops back to back
entrypoint:
  ref: a
inputs:
  first: 1
  second: 2
  list: [1, 2, 3]

actions:
  - ref: a
    action: core.transform.reshape
    for_each: ${{ for var.x in INPUTS.list }}
    args:
      value: ${{ FN.add(var.x, INPUTS.first) }}
  - ref: b
    action: core.transform.reshape
    depends_on:
      - a
    for_each: ${{ for var.x in ACTIONS.a.result }}
    args:
      value: ${{ FN.add(INPUTS.second, var.x) }}
