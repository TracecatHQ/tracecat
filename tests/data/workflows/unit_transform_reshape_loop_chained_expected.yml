ACTIONS:
  a:
    result:
      - 2
      - 3
      - 4
    result_typename: "list"
  b:
    result:
      - 4
      - 5
      - 6
    result_typename: "list"

INPUTS:
  first: 1
  second: 2
  list: [1, 2, 3]
TRIGGER:
